pkg/executors/docker_compose_executor.go

```diff
@@ -502,13 +504,14 @@ func (e *DockerComposeExecutor) pullDockerImages() int {
 
 	if err := cmd.Wait(); err != nil {
 		log.Println("Docker pull failed", err)
+		e.SubmitStats("semphoreci/android", "error.rate", []string{"android"}, 1)
 		exitCode = 1
 	}
 
 	log.Println("Docker pull finished. Exit Code", exitCode)
 
 	commandFinishedAt := int(time.Now().Unix())
-
+	e.SubmitStats("semphoreci/android", "pull.time", []string{"android"}, (commandFinishedAt - commandStartedAt))
```

Comment:
This part of the code is too specific for this level. Please enclose it in the fucntion:

Please refactor it to:

```
e.SubmitDockerPullTime()
```

And the function to:

```
func (e *DockerComposeExecutor) SubmitDockerPullTime(duration int) {
  if strings.Contains(e.jobRequest.Compose.Containers[0].Image, "semaphore/android") {
     // only submiting android metrics.
     watchman.SubmitWithTags("semphoreci/android", ....
  }
}
```

Reply:
Construct names in a way that allows adding new namespaces.

The following would be a good namespace.

```
compose.docker.pull.duration, tags { semphoreci/android }
```